java-frontend/src/main/java/org/sonar/java/regex/RegexParser.java

```diff
@@ -537,17 +539,22 @@ private RegexTree parseCharacterClass() {
       expected("']'");
     }
     IndexRange range = openingBracket.getRange().extendTo(characters.getCurrentStartIndex());
-    return new CharacterClassTree(source, range, negated, contents);
+    return new CharacterClassTree(source, range, openingBracket, negated, contents);
```

Comment:
I need the opening bracket in order to report a "+1" secondary only on the bracket (reporting on the whole character class can lead to overlapping secondaries if something inside the class has a "+n" as well and, as we say, overlapping secondaries lead to ugly output in SonarQube). I don't need the closing bracket for anything, so that's why it's not there.

I agree that it'd be more consistent to have both, but I feel like it's a similar thing to having `before` and `after`: only add both if and when you need both of them.